### ISSUE TITLE
CastError: Cast to ObjectId failed for value "" at path "_id"

### ISSUE THREAD

**Issue by jalalat** [apollographql/apollo-server#556]
I switched from using express-graphql (with Relay Modern) to apollo-server. I am getting this error

> CastError: Cast to ObjectId failed for value "" at path "_id"

on my schema:

```
import mongoose from 'mongoose';
const Schema = mongoose.Schema;

const PostSchema = new Schema({
  _id: { type: Schema.Types.ObjectId },
  title: { type: String },
  details: { type: String },
  views: { type: Number, default: 0 },
  likes: { type: Number, default: 0 },
  dislikes: { type: Number, default: 0 },
  createdAt: { type: Date },
});
```

If i change the _id: { type: Schema.Types.ObjectId } to **_id: { type: String }** it works. 

Edit: i found the reason. Initial value for making a call to this schema (through query was "" i.e. empty) and that's why it was giving me this error. However i do not get this error if i switch to express-graphql (express-graphql: "^0.6.7")

In conclusion string value does not get converted to ObjectId in apollo-server (apollo-server-express: "^1.1.2")

**Comment by bogdanv28** [comment]
your answer was helpful, i forgot to put "await" when calling a fct to get the id, and the result was an empty string.
